django_x509/base/mixin.py

```diff
@@ -0,0 +1,19 @@
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
+from django.utils.translation import ugettext_lazy as _
+
+from django_x509 import settings as app_settings
+
+
+class CrlDownloadMixin:
+    def crl_view(self, request, pk):
+        authenticated = request.user.is_authenticated
+        authenticated = authenticated() if callable(authenticated) else authenticated
+        if app_settings.CRL_PROTECTED or not authenticated:
+            return HttpResponse(_('Forbidden'), status=403, content_type='text/plain')
+        instance = get_object_or_404(self.model, pk=pk)
+        response = HttpResponse(
+            instance.crl, status=200, content_type='application/x-pem-file'
+        )
+        response['Content-Disposition'] = f'attachment; filename={pk}.crl'
```

Comment:
Why did you add this? In the previous code we didn't force the download.

Reply:
When this function is triggered by the API endpoint, then the file name doesn't get's added automatically instead the file in `crl`format get's downloaded without any name, Will try to implement it once again and see if I can do anything from the API part.

Reply:
ok, let's avoid changing the functionality here for reasons of backward compatibility.